src/Magnum/Shaders/FlatGL.cpp

```diff
@@ -277,22 +289,44 @@ template<UnsignedInt dimensions> FlatGL<dimensions>::FlatGL(const Flags flags
     /* Set defaults in OpenGL ES (for desktop they are set in shader code itself) */
     #ifdef MAGNUM_TARGET_GLES
     #ifndef MAGNUM_TARGET_GLES2
-    if(flags >= Flag::UniformBuffers) {
+    if(_flags >= Flag::UniformBuffers) {
         /* Draw offset is zero by default */
     } else
     #endif
     {
         setTransformationProjectionMatrix(MatrixTypeFor<dimensions, Float>{Math::IdentityInit});
-        if(flags & Flag::TextureTransformation)
+        if(_flags & Flag::TextureTransformation)
             setTextureMatrix(Matrix3{Math::IdentityInit});
         /* Texture layer is zero by default */
         setColor(Magnum::Color4{1.0f});
-        if(flags & Flag::AlphaMask) setAlphaMask(0.5f);
+        if(_flags & Flag::AlphaMask) setAlphaMask(0.5f);
         /* Object ID is zero by default */
     }
     #endif
 }
 
+template<UnsignedInt dimensions> FlatGL<dimensions>::FlatGL(Flags flags
+    #ifndef MAGNUM_TARGET_GLES2
+    , UnsignedInt materialCount, UnsignedInt drawCount
+    #endif
+):
+    #ifndef MAGNUM_TARGET_GLES2
+    FlatGL(compile(flags, materialCount, drawCount))
+    #else
+    FlatGL{compile(flags)}
+    #endif
+{}
```

Comment:
This corresponds to two separate constructors in the header. Below is another one, that forwards (Flags) to (Flags, Int, Int). You might have placed them here on purpose to not clutter the header, but I can move them both if needed.

Reply:
Ah, excellent point :+1: Yep, it would probably make sense to move both, since both are now just thin wrappers that eventually delegate to `compile()`.

(Just to be clear, inline them directly inside the class definition. But I guess you know what I meant ;))